crates/chain/src/tx_graph.rs

```diff
@@ -451,6 +451,21 @@ impl<A> TxGraph<A> {
     }
 }
 
+impl<A: Clone + Ord> TxGraph<A> {
+    /// Transform the [`TxGraph`] to have [`Anchor`]s of another type.
+    ///
+    /// This takes in a closure of signature `FnMut(A) -> A2` which is called for each [`Anchor`] to
+    /// transform it.
+    pub fn map_anchors<A2: Clone + Ord, F>(self, f: F) -> TxGraph<A2>
+    where
+        F: FnMut(A) -> A2,
+    {
+        let mut new_graph = TxGraph::<A2>::default();
+        new_graph.apply_changeset(self.initial_changeset().map_anchors(f));
```

Comment:
There are multiple anchor-related fields in `TxGraph`, and only one in `ChangeSet`.  The function caller passed in possible non-determined, so it is good to map it once here to avoid issues from a non-determined function.

Reply:
Huh isn't there only one acnhor related field on `TxGraph` also?

Reply:
> Huh isn't there only one acnhor related field on `TxGraph` also?

We also have anchors in `TxNode` which should have 1 to 1 mapping to the `anchors` field. I think the implementation right now is safer and easier to read. If in the future, there is clear performance problems we can do something more efficient.